commit 769094bb66a408181e7a79a2bb37927280cd9aed
Author: Mike Bayer <mike_mp@zzzcomputing.com>
Date:   Wed Dec 13 10:32:50 2017 -0500

    Add Nova max_concurrent_builds, rpc response timeout to undercloud conf
    
    Running multiple-controller/compute overcloud builds on virtualized
    hardware frequently leads to the issue that Nova hangs up
    during spawning of instances.   Reducing the number of
    concurrent spawns to a low number like 2 as well as increasing
    nova and ironic rpc response time to a value like 600 typically
    resolves this issue; this patch allows the parameters to be part of the
    undercloud conf extra parameters.
    
    Change-Id: Ib46fd4cf82dc587bd92e0d1a6f221006ca922078

diff --git a/roles/undercloud-deploy/templates/hieradata-overrides-classic-undercloud.yaml.j2 b/roles/undercloud-deploy/templates/hieradata-overrides-classic-undercloud.yaml.j2
index b478342..ee54f07 100644
--- a/roles/undercloud-deploy/templates/hieradata-overrides-classic-undercloud.yaml.j2
+++ b/roles/undercloud-deploy/templates/hieradata-overrides-classic-undercloud.yaml.j2
@@ -67,6 +67,21 @@ nova::api::osapi_compute_workers: "%{hiera('undercloud_workers')}"
 nova::conductor::workers: "%{hiera('undercloud_workers')}"
 {% endif %}
 
+{% if undercloud_nova_max_concurrent_builds is defined %}
+nova::compute::ironic::max_concurrent_builds: {{undercloud_nova_max_concurrent_builds}}
+{% endif %}
+
+{% if undercloud_nova_rpc_response_timeout is defined %}
+nova::rpc_response_timeout: {{undercloud_nova_rpc_response_timeout}}
+{% endif %}
+
+{% if undercloud_ironic_rpc_response_timeout is defined %}
+ironic::rpc_response_timeout: {{undercloud_ironic_rpc_response_timeout}}
+{% endif %}
+
+
+
+
 {% if undercloud_enable_novajoin or enable_tls_everywhere %}
 nova::api::vendordata_dynamic_connect_timeout: {{ novajoin_connect_timeout }}
 nova::api::vendordata_dynamic_read_timeout: {{ novajoin_read_timeout }}
diff --git a/roles/undercloud-deploy/templates/hieradata-overrides-t-h-t-undercloud.yaml.j2 b/roles/undercloud-deploy/templates/hieradata-overrides-t-h-t-undercloud.yaml.j2
index 1ebe08a..79cea08 100644
--- a/roles/undercloud-deploy/templates/hieradata-overrides-t-h-t-undercloud.yaml.j2
+++ b/roles/undercloud-deploy/templates/hieradata-overrides-t-h-t-undercloud.yaml.j2
@@ -61,6 +61,19 @@ parameter_defaults:
     nova::conductor::workers: "%{hiera('undercloud_workers')}"
     {% endif %}
 
+    {% if undercloud_nova_max_concurrent_builds is defined %}
+    nova::compute::ironic::max_concurrent_builds: {{undercloud_nova_max_concurrent_builds}}
+    {% endif %}
+
+    {% if undercloud_nova_rpc_response_timeout is defined %}
+    nova::rpc_response_timeout: {{undercloud_nova_rpc_response_timeout}}
+    {% endif %}
+
+    {% if undercloud_ironic_rpc_response_timeout is defined %}
+    ironic::rpc_response_timeout: {{undercloud_ironic_rpc_response_timeout}}
+    {% endif %}
+
+
     {% if undercloud_swift_workers is defined %}
     swift::proxy::workers: {{undercloud_swift_workers}}
     {% elif undercloud_workers is defined %}
